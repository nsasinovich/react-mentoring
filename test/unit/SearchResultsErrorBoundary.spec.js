import { shallow, mount } from 'enzyme';
import React from 'react';
import SearchResultsErrorBoundary from 'src/components/search_results_list/search_results_error_boundary/search_results_error_boundary';

describe('<SearchResultsErrorBoundary />', () => {
    it('should render SearchResultsErrorBoundary and match snapshot', () => {
        const wrapper = shallow(<SearchResultsErrorBoundary>
            <div>Test</div>
        </SearchResultsErrorBoundary>);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render SearchResultsErrorBoundary with error', () => {
        const wrapper = shallow(<SearchResultsErrorBoundary>
            <div>Test</div>
        </SearchResultsErrorBoundary>);
        wrapper.setState({ hasError: true });

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.state().hasError).toEqual(true);
        expect(wrapper.containsMatchingElement(<h1 className="error-boundary">Something went wrong.</h1>)).toBe(true);
    });

    it('should catch errors with componentDidCatch', () => {
        jest.spyOn(SearchResultsErrorBoundary.prototype, 'componentDidCatch');

        const wrapper = mount(<SearchResultsErrorBoundary>
            <div>Test</div>
        </SearchResultsErrorBoundary>);
        wrapper.instance().componentDidCatch();

        expect(wrapper.state().hasError).toEqual(true);
        expect(SearchResultsErrorBoundary.prototype.componentDidCatch).toHaveBeenCalled();
    });
});
