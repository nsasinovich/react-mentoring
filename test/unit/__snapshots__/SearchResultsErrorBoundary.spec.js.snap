// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchResultsErrorBoundary /> should render SearchResultsErrorBoundary and match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchResultsErrorBoundary>
    <div>
      Test
    </div>
  </SearchResultsErrorBoundary>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Test",
    },
    "ref": null,
    "rendered": "Test",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Test",
      },
      "ref": null,
      "rendered": "Test",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<SearchResultsErrorBoundary /> should render SearchResultsErrorBoundary with error 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchResultsErrorBoundary>
    <div>
      Test
    </div>
  </SearchResultsErrorBoundary>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Something went wrong.",
      "className": "error-boundary",
    },
    "ref": null,
    "rendered": "Something went wrong.",
    "type": "h1",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Something went wrong.",
        "className": "error-boundary",
      },
      "ref": null,
      "rendered": "Something went wrong.",
      "type": "h1",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
